src/modules/differential_drive_control/DifferentialDriveKinematics/DifferentialDriveKinematics.hpp

```diff
@@ -31,29 +31,37 @@
  *
  ****************************************************************************/
 
-#include "ActuatorEffectivenessRoverDifferential.hpp"
-#include <ControlAllocation/ControlAllocation.hpp>
+#pragma once
 
-using namespace matrix;
+#include <matrix/matrix/math.hpp>
 
-bool
-ActuatorEffectivenessRoverDifferential::getEffectivenessMatrix(Configuration &configuration,
-		EffectivenessUpdateReason external_update)
+class DifferentialDriveKinematics
 {
-	if (external_update == EffectivenessUpdateReason::NO_EXTERNAL_UPDATE) {
-		return false;
-	}
-
-	configuration.addActuator(ActuatorType::MOTORS, Vector3f{0.f, 0.f, 0.5f}, Vector3f{0.5f, 0.f, 0.f});
-	configuration.addActuator(ActuatorType::MOTORS, Vector3f{0.f, 0.f, -0.5f}, Vector3f{0.5f, 0.f, 0.f});
-	_motors_mask = (1u << 0) | (1u << 1);
-	return true;
-}
-
-void ActuatorEffectivenessRoverDifferential::updateSetpoint(const matrix::Vector<float, NUM_AXES> &control_sp,
-		int matrix_index, ActuatorVector &actuator_sp, const matrix::Vector<float, NUM_ACTUATORS> &actuator_min,
-		const matrix::Vector<float, NUM_ACTUATORS> &actuator_max)
-{
-	stopMaskedMotorsWithZeroThrust(_motors_mask, actuator_sp);
-}
+public:
+	DifferentialDriveKinematics() = default;
+	~DifferentialDriveKinematics() = default;
+
+	void computeInverseKinematics();
+	void computeForwardsKinematics();
```

Comment:
I would remove this yet unused method for now. Better bring it in when it's actually used (or at least logged).

Reply:
Agreed, removed.